Coutoire: Add Color Annotations

kjellr opened the pull request:
Follows the mold of #1944 to add basic color annotations for Coutoire. 

![Screen Shot 2020-04-30 at 16 38 32](https://user-images.githubusercontent.com/1202812/80756743-10b8dc00-8b01-11ea-9b0b-07d3b8ad2de3.png)


Still needs: 

- [x] Refinement around contrast. 
- [x] Double checking mobile styles.
- [x] Add featured color palettes.

kjellr committed:
Coutoire: Add Color Annotations

kjellr committed:
Fix a copy/paste error.

kjellr committed:
Remove color contrast rules.

kjellr committed:
Add grays

kjellr committed:
Add featured color palettes

allancole committed:
Coutoire: Recompile CSS to remove !important from utility classes, update npm dependency.

allancole committed:
Coutoire: Connect editor color palette to color annotations.

allancole committed:
Coutoire: Revise conditionals to fix array offset issue, cody tidying.

allancole committed:
Coutoire: Remove `!important` flags from utility classes.

kjellr committed:
Add editor color annotations.

allancole committed:
Coutoire: Fix txt/bg text color.

allancole committed:
Coutoire: Fix nested button text-color, code tidying.

allancole committed:
Coutoire: Fix rtl nested button text-color, code tidying.

ianstewart commented:
I tested the dark colour scheme for Coutoire this morning.

- The editor was **not** picking up the dark colour scheme.
- The front-end and Editor previews worked correctly.

allancole committed:
Coutoire: Fix color-annotations selector specificity for Cover and Button Block.

allancole committed:
Coutoire: Fix editor color-annotations selector specificity for Cover and Button Block.

- Change `black` color name to `#000000` hex equivalent.
- Change `white` color name to `#FFFFFF` hex equivalent.

allancole committed:
Coutoire: Renaming css color names to their hex equivalents.

allancole committed:
Coutoire: Add additional colors to default palette to sync with parent theme

allancole committed:
Coutoire: Fix rtl color-annotations selector specificity for Cover and Button Block.

allancole committed:
Coutoire: Improve editor annotations

- Replace color names with hex equivalents
- Remove text-color initial rules from background-color utility classes to allow color options to work more explicitly
- Clean up for nested link colors

allancole committed:
Coutoire: Restore editor wrapper selectors.

- Fixes default button and nested text colors

allancole committed:
Coutoire: Replace CSS color names with hex equivalents

kjellr commented:
Found a handful of issues this afternoon: 

- [ ] Default cover blocks do not match up between the front and back end.

Editor|Front End
---|---
![Screen Shot 2020-06-05 at 2 52 06 PM](https://user-images.githubusercontent.com/1202812/83913036-dfa17c00-a73c-11ea-974b-60810c03e7cf.png)|![Screen Shot 2020-06-05 at 2 52 13 PM](https://user-images.githubusercontent.com/1202812/83913038-e29c6c80-a73c-11ea-8267-95329a63073b.png)

- [ ] Buttons do not match up between the front and back end.

Editor|Front End
---|---
![Screen Shot 2020-06-05 at 2 59 09 PM](https://user-images.githubusercontent.com/1202812/83913205-2b542580-a73d-11ea-8dfc-e368e35a167a.png)|![Screen Shot 2020-06-05 at 2 58 40 PM](https://user-images.githubusercontent.com/1202812/83913209-2db67f80-a73d-11ea-93a3-212c6a7db667.png)

- [ ] Services Template: Grid appears unreadable. 

Editor|Front End
---|---
![Screen Shot 2020-06-05 at 3 01 42 PM](https://user-images.githubusercontent.com/1202812/83913470-7a9a5600-a73d-11ea-852f-ef92cb112ef5.png)|![Screen Shot 2020-06-05 at 3 00 39 PM](https://user-images.githubusercontent.com/1202812/83913476-7d954680-a73d-11ea-9428-1aa675928ed4.png)

- [ ] Portfolio Template: Image Slider appears with white bars on the sides. 

![Screen Shot 2020-06-05 at 3 03 44 PM](https://user-images.githubusercontent.com/1202812/83913697-d1a02b00-a73d-11ea-8455-e54046603ef1.png)

allancole committed:
Coutoire: Cleanup cover, button, table block annotations

allancole committed:
Coutoire: Cleanup cover, button, utility annotations

allancole committed:
Coutoire: Use hex colors instead of color names.

allancole committed:
Coitoire: Add table styles and clean up utilities.

allancole committed:
Coutoire: Add table block support.

allancole committed:
Coutoire: Cleanup table block color-annotations

allancole committed:
Coutoire: Cleanup table block color-annotations

allancole committed:
Coutoire: Clean up dim support in cover block.

allancole committed:
Coutoire: Clean up dim support in cover block.

allancole committed:
Coutoire: Clean up dim support in cover block.

allancole committed:
Coutoire: Clean up dim support in cover block.

allancole committed:
Coutoire: Recompile styles.

- Clean up dim support in cover block.
- Replace colornames with hex equivalents

allancole commented:
Merging this PR and syncing to wpcom.
